src/bp-core/profile-search/bps-xprofile.php

```diff
@@ -673,20 +673,45 @@ function bp_ps_anyfield_search( $f ) {
 
 	}
 
+	if ( in_array( $filter, array( 'contains', 'like' ) ) ) {
+		$every_word_clauses = array();
+		$query_placeholder  = array();
```

Comment:
$query_placeholder = array();
Move this variable before `if ( in_array( $filter, array( 'contains', 'like' ) ) ) {` this condition